src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -292,26 +292,7 @@ private void LogScriptIfGiven(string type, string scriptPath)
         /// <returns>The maximum version that satisfies the requested version spec.</returns>
         private string GetMatchingTargetVersion(IProgrammingPlatform platform, string targetVersionSpec)
```

Comment:
This api is used when an explicit version is provided by a user. Since the version provided by an end user can be just major or major.minor or major.minor.patch, this api tries to resolve it. Note that we do a similar thing in each of the platform detectors too...so there the use case is that an explicit platform was not provided but instead a version was detected and that could also be just major or major.minor or major.minor.patch and we need to resolve it.

Reply:
can you add  some tests for this? ignore if they are already there

> This api is used when an explicit version is provided by a user. Since the version provided by an end user can be just major or major.minor or major.minor.patch, this api tries to resolve it. Note that we do a similar thing in each of the platform detectors too...so there the use case is that an explicit platform was not provided but instead a version was detected and that could also be just major or major.minor or major.minor.patch and we need to resolve it.

Reply:
So there are several tests within `DefaultBuildScriptGeneratorTest` do the exact things for testing but I'm not sure if they covered all the cases.

Reply:
@kichalla could you check if all cases in the scenarios are covered in the tests?

Reply:
> can you add some tests for this? ignore if they are already there

yes, we already have several tests for these.

> @kichalla could you check if all cases in the scenarios are covered in the tests?

We currently have several tests for these but they are not necessarily as unit tests under `DefaultBuildScriptGeneratorTest`. We have quite a few covered by `SemantingVersioningResolver`, each platform's detector test and integration tests too.